sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -634,4 +650,31 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       },
     });
   }
+
+  /**
+   * Check if the given institution has already a program with the given SABC code.
+   * @param institutionId id of the institution to have the programs retrieved.
+   * @param sabcCode SABC code.
+   * @returns true in case it has already a SABC code for the institution.
+   */
+  async hasExistingProgramWithSameSABCCode(
+    institutionId: number,
+    sabcCode: string,
+    programId: number,
+  ): Promise<boolean> {
+    const result = await this.repo.findOne({
+      select: {
+        id: true,
+      },
+      where: {
+        sabcCode: sabcCode,
+        institution: {
+          id: institutionId,
+        },
+        ...(programId && { id: Not(Equal(programId)) }),
```

Comment:
mm... what is happing here?

Reply:
if a programId is passed, add a where condition to exclude the program. That is in case it is an update for the program form.

Reply:
I would suggest a syntax like below. I believe that it is more aligned with what we did in the past. Can you check it, please?
![image](https://user-images.githubusercontent.com/61259237/204637433-1919b3aa-40bd-4295-b087-b92c914d7b66.png)

Reply:
yes, I see... ok. Thanks!